posthog/test/test_middleware.py

```diff
@@ -8,15 +12,28 @@ class TestSignup(TestCase):
     def setUp(self):
         super().setUp()
         self.client = Client()
+        self.team: Team = Team.objects.create(api_token="tokenABC123")
 
     def test_ip_range(self):
```

Comment:
Now that this is making a real request, you need to patch process_event
```
@patch("posthog.tasks.process_event.process_event.delay")
```

alternatively just send a bad request, the point of this test isn't that you can make correct requests, the point is to test that you are allowed to hit that endpoint

Reply:
Thanks Tim! I'm doing the bad request option to have better separation of concerns in the tests. Seems to be working now.